lib/src/model/tile_model.dart

```diff
@@ -35,34 +38,47 @@ class TileFeature {
     }
 
     final geometry = _geometry;
-    if (geometry != null) {
+    if (geometry != null && _points == null) {
       _points = decodePoints(geometry).toList(growable: false);
-      _geometry = null;
     }
 
-    return _points;
+    return _points ?? [];
   }
 
-  List<Path> get paths {
+  List<Path> getPaths({List<double> dashLengths = const []}) {
     if (type == TileFeatureType.point) {
       throw StateError('Cannot get paths from a point feature');
     }
 
     final geometry = _geometry;
     if (geometry != null) {
-      // ignore: missing_enum_constant_in_switch
-      switch (type) {
-        case TileFeatureType.linestring:
-          _paths = decodeLineStrings(geometry).toList(growable: false);
-          break;
-        case TileFeatureType.polygon:
-          _paths = decodePolygons(geometry).toList(growable: false);
-          break;
+      if (_paths == null) {
+        // ignore: missing_enum_constant_in_switch
+        switch (type) {
+          case TileFeatureType.linestring:
+            _paths = decodeLineStrings(geometry).toList(growable: false);
+            break;
+          case TileFeatureType.polygon:
+            _paths = decodePolygons(geometry).toList(growable: false);
+            break;
+        }
       }
-      _geometry = null;
     }
 
-    return _paths;
+    if (dashLengths.length >= 2) {
+      if (_dashedPaths.containsKey(dashLengths)) {
```

Comment:
Based on our previous discussion, would it make sense to remove `_dashedPaths`? The lifecycle of a tile could be quite long, and unless I'm mistaken the key (a list of doubles) could vary a lot.

Alternatively, we could limit the size of `_dashedPaths` (e.g. to some arbitrary size, e.g. 20)
In that case, to avoid looking up the key in the map twice, consider something like this:

```dart
var path = _dashedPaths[dashLengths];
if (path == null) {
  path = _paths?.map ...
  _dashedPaths[dashLengths] = path;
}
return path;
```

Reply:
Hey, yes i think it makes sense to delete this completely. I've removed it now.